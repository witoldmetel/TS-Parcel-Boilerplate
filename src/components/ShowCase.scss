.showcase {
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  color: #fff;
  padding: 0 20px;

  .video-container {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    overflow: hidden;

    &:after {
      content: '';
      z-index: 1;
      height: 100%;
      width: 100%;
      top: 0;
      left: 0;
      background: rgba(0, 0, 0, 0.5);
      position: absolute;
    }

    video {
      min-width: 100%;
      min-height: 100%;
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
      object-fit: cover;
    }
  }

  .content {
    z-index: 2;

    h1,
    h3 {
      color: white;
      font-weight: 500;
    }

    .btn {
      display: inline-block;
      padding: 10px 30px;
      background: var(--primary-color);
      color: #fff;
      border-radius: 5px;
      border: solid #fff 1px;
      margin-top: 25px;
      opacity: 0.7;

      &:hover {
        transform: scale(0.98);
      }
    }
  }
}
