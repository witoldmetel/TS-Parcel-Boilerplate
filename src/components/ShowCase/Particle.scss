.showcase {
  .particles-container {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    overflow: hidden;

    &:after {
      content: '';
      z-index: 1;
      height: 100%;
      width: 100%;
      top: 0;
      left: 0;
      background: rgba(0, 0, 0, 0.4);
      position: absolute;
    }
  }
}
