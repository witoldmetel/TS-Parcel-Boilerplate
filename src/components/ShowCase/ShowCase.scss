.showcase {
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  color: #fff;
  padding: 0 20px;

  .particles-container {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    overflow: hidden;

    &:after {
      content: '';
      z-index: 1;
      height: 100%;
      width: 100%;
      top: 0;
      left: 0;
      background-attachment: fixed;
      background-position: right;
      background-size: cover;
      background-image: url('../../assets/graphics/background_img.jpg');
      position: absolute;
      opacity: 0.75;
    }
  }

  .content {
    z-index: 2;

    h1,
    h2 {
      color: var(--secondary-color);
      font-weight: 500;
      -webkit-text-stroke: 1px black;
    }

    .btn {
      display: inline-block;
      padding: 10px 30px;
      background: var(--primary-color);
      color: var(--text-color);
      border-radius: 5px;
      border: solid var(--text-color) 1px;
      margin-top: 25px;
      text-decoration: none;
      opacity: 0.85;

      &:hover {
        p {
          transition-duration: 1s;
          transform: scale(1.3);
        }
      }

      p {
        font-weight: bold;
        margin: 0;
      }
    }
  }
}
