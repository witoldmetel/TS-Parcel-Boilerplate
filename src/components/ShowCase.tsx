import React, { Component } from 'react';

import intro from '../assets/intro.mp4';

import './ShowCase.scss';

export class ShowCase extends Component {
  render() {
    return (
      <div className="showcase">
        <div className="video-container">
          <video autoPlay muted loop>
            <source src={intro} type="video/mp4" />
          </video>
        </div>
        <div className="content">
          <h1>React + TypeScript + Parcel</h1>
          <h3>Simple Boilerplate</h3>
          <a href="#about" className="btn">
            Read More
          </a>
        </div>
      </div>
    );
  }
}
