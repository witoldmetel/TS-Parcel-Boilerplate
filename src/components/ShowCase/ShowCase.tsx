import React, { Component } from 'react';

import './ShowCase.scss';

export class ShowCase extends Component {
  render() {
    return (
      <div className="showcase">
        <div className="gif-container">
          <img src="https://thumbs.gfycat.com/ImpishUnequaledIchneumonfly-size_restricted.gif" />
        </div>
        <div className="content">
          <h1>React + TypeScript + Parcel</h1>
          <h3>Simple Boilerplate</h3>
          <a href="#about" className="btn">
            Read More
          </a>
        </div>
      </div>
    );
  }
}
