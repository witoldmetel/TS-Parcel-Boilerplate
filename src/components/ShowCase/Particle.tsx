import React, { Component } from 'react';
import p5 from 'p5';

import './Particle.scss';

export class Particle extends Component {
  private particlesRef: React.RefObject<HTMLInputElement>;

  constructor(props: any) {
    super(props);

    this.particlesRef = React.createRef();
  }

  componentDidMount() {
    new p5((particle) => {
      // Init playground for P5 circles
      particle.setup = () => {
        particle.createCanvas(window.innerWidth, window.innerHeight).parent(this.particlesRef.current);
      };

      // Circles position
      const position = particle.createVector(
        particle.random(0, window.innerWidth),
        particle.random(0, window.innerHeight),
      );

      // Circles size
      const size = 10;

      particle.draw = () => {
        particle.noStroke();
        particle.fill('rgba(255,255,255,0.5');
        particle.circle(position.x, position.y, size);
      };
    });
  }

  render() {
    return <div className="particles-container" ref={this.particlesRef}></div>;
  }
}
