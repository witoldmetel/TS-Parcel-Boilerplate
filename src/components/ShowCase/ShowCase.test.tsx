import * as React from 'react';
import { shallow } from 'enzyme';

import { ShowCase } from './ShowCase';

describe('<ShowCase>', () => {
  it('renders without crashing', () => {
    shallow(<ShowCase />);
  });

  it('renders correctly', () => {
    const wrapper = shallow(<ShowCase />);

    expect(wrapper).toMatchSnapshot();
  });
});
