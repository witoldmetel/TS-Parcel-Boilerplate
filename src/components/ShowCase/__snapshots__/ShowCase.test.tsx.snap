// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowCase> renders correctly 1`] = `
<div
  className="showcase"
>
  <div
    className="particles-container"
  >
    <Particle />
  </div>
  <div
    className="content"
  >
    <h1>
      React + TypeScript + Parcel
    </h1>
    <h2>
      Simple Boilerplate
    </h2>
    <a
      className="btn"
      href="#about"
    >
      Read More
    </a>
  </div>
</div>
`;
